### PR TITLE
[DOCS] Add public API docstring for `expect_column_values_to_be_between` `validate_configuration`

### PR DESCRIPTION
**Pull request by NathanFarmer**
Changes proposed in this pull request:
- DX-237
- Add docstring

### Definition of Done
- [X] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [X] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [X] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: great_expectations/expectations/core/expect_column_values_to_be_between.py [hunk 1]
```diff
@@ -235,20 +237,29 @@ class ExpectColumnValuesToBeBetween(ColumnMapExpectation):
...
-        """
-        Validates that a configuration has been set, and sets a configuration if it has yet to be set. Ensures that
-        necessary configuration arguments have been provided for the validation of the expectation.
+        """Validates the configuration of an Expectation.
+
+        For `expect_column_values_to_be_between` it is required that:
+            - One of `min_value` or `max_value` is not `None`.
+            - `min_value` and `max_value` are one of the following types: `datetime`, `float`, `int`, or `dict`
+            - If `min_value` or `max_value` is a `dict`, it is assumed to be an Evaluation Parameter, and therefore the
+              dictionary keys must be `$PARAMETER`.
```
Comment (Shinnnyshinshin) [inline great_expectations/expectations/core/expect_column_values_to_be_between.py]: A small nit, but I think the rendering will be more consistent with other files if it was un-tabbed to be in line with the `For`. 

Screen Shot 2023-01-20 at 1 17 13 PM

Looks like [this](https://docs.greatexpectations.io/docs/reference/api/great_expectations/util/)

**Review by Shinnnyshinshin — CHANGES_REQUESTED** [review]
Requesting a very small change with the list formatting

File: great_expectations/expectations/core/expect_column_values_to_be_between.py (hunk 1 above)
Comment (NathanFarmer) [reply great_expectations/expectations/core/expect_column_values_to_be_between.py]: ✅

**Review by anthonyburdi — APPROVED** [review]
Nice! Like the better error.

**Pull request merged**